src/kudu/consensus/consensus_queue.cc

```diff
@@ -1025,6 +1050,10 @@ void PeerMessageQueue::EndWatchForSuccessor() {
   tl_filter_fn_ = nullptr;
 }
 
+Status PeerMessageQueue::GetNextRoutingHopFromLeader(const string& dest_uuid, string* next_hop) const {
+  return routing_table_->NextHop(local_peer_pb_.permanent_uuid(), dest_uuid, next_hop);
```

Comment:
does this not need protection by lock? The lifetime of PeerMessageQueue should involve multiple changes to routing_table_?

Reply:
No, routing_table_ is set only once (at construction time) and it's a shared_ptr to an instance of DurableRoutingTable, which is thread-safe. DRT lives forever (as long as ConsensusMetadataManager lives) or until it's deleted (in kuduraft it's never deleted, but that would only happen when destroying the tablet including the consensus instance).